mne_connectivity/spectral.py

```diff
@@ -916,19 +920,58 @@ def spectral_connectivity(data, method='coh', indices=None, sfreq=2 * np.pi,
                                 this_con_flat.shape[1:],
                                 dtype=this_con_flat.dtype)
             this_con[indices_use] = this_con_flat
+
+            # ravel 2D connectivity into a 1D array
+            # while keeping other dimensions
+            this_con = this_con.reshape((n_signals ** 2,) +
+                                        this_con_flat.shape[1:])
             con.append(this_con)
+    # number of nodes in the original data,
+    n_nodes = n_signals
+
+    # create a list of connectivity containers
+    conn_list = []
+    for _con in con:
+        # create the connectivity container
+        if mode in ['multitaper', 'fourier']:
+            # spectral only
+            conn = SpectralConnectivity(
+                data=_con,
+                names=names,
+                freqs=freqs,
+                method=method,
+                n_nodes=n_nodes,
+                spec_method=mode,
+                indices=indices,
+                n_epochs_used=n_epochs,
+                freqs_used=freqs_used,
+                times_used=times,
+                n_tapers=n_tapers,
+            )
+        elif mode == 'cwt_morlet':
+            # spectrotemporal
+            conn = SpectroTemporalConnectivity(
+                data=_con,
+                names=names,
+                freqs=freqs,
+                times=times,
+                n_nodes=n_nodes,
+                method=method,
+                spec_method=mode,
+                indices=indices,
+                n_epochs_used=n_epochs,
+                freqs_used=freqs_used,
+                n_tapers=n_tapers
+            )
```

Comment:
You can DRY this code (and have better/clearer control flow) with:
```
kwargs = dict(data=_con, ...)
if mode in ('multitaper', 'fourier'):
    klass = SpectralConnectivity
    kwargs.update(times_used=times, ...)
    ...
else:
    assert mode == 'cwt_morlet'
    klass = SpectroTemporalConnectivity
    kwargs.update(times=times, ...)
conn_list.append(klass(**kwargs))
```
it makes it immediately clear what is shared (at the beginning) and what is different (in the branches). Also use of if/else rather than if/elif makes it clear that one branch will *always* be entered, and the `assert` just asserts something that we should know from previous checks above -- I think of these like "active comments" in the sense that it both says "this should be the case" and makes sure it is.